build_web_compilers/lib/src/web_entrypoint_builder.dart

```diff
@@ -311,20 +312,26 @@ class WebEntrypointBuilder implements Builder {
             entrypointExtension: compiler.extension,
           ));
         case WebCompiler.Dart2Wasm:
-          compilationSteps.add(bootstrapDart2Wasm(
-              buildStep, compiler.compilerArguments, compiler.extension));
+          compilationSteps.add(Future(() async {
+            dart2WasmResult = await bootstrapDart2Wasm(
+                buildStep, compiler.compilerArguments, compiler.extension);
+          }));
       }
     }
     await Future.wait(compilationSteps);
-    if (_generateLoader(buildStep.inputId) case (var id, var loader)?) {
+    if (_generateLoader(buildStep.inputId, dart2WasmResult)
+        case (var id, var loader)?) {
       await buildStep.writeAsString(id, loader);
     }
   }
 
-  (AssetId, String)? _generateLoader(AssetId input) {
+  (AssetId, String)? _generateLoader(
+      AssetId input, Dart2WasmBootstrapResult? dart2WasmResult) {
     var loaderExtension = options.loaderExtension;
     var wasmCompiler = options.optionsFor(WebCompiler.Dart2Wasm);
-    if (loaderExtension == null || wasmCompiler == null) {
+    if (loaderExtension == null ||
+        wasmCompiler == null ||
+        dart2WasmResult == null) {
```

Comment:
This looks like it will now do nothing in cases where previously it might have done something? When `dart2WasmResult` is null and the others are not null?

Reply:
Makes sense, thanks :)